opencensus/exporters/trace/zipkin/zipkin_exporter.h

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENCENSUS_EXPORTERS_TRACE_ZIPKIN_ZIPKIN_EXPORTER_H_
+#define OPENCENSUS_EXPORTERS_TRACE_ZIPKIN_ZIPKIN_EXPORTER_H_
+
+#include <memory>
+#include <string>
+
+#include "absl/memory/memory.h"
+#include "absl/time/time.h"
+#include "opencensus/trace/exporter/span_exporter.h"
+#include "opencensus/trace/span.h"
+
+namespace opencensus {
+namespace exporters {
+namespace trace {
+
+// MessageCodec is used for encoding message sets.
+class MessageCodec {
+ public:
+  virtual ~MessageCodec() = default;
+
+  virtual const std::string name() const = 0;
+
+  virtual std::string Encode(
+      const std::vector<::opencensus::trace::exporter::SpanData> &spans) = 0;
+};
+
+// JSON encoding
+class JsonCodec : public MessageCodec {
+ public:
+  const std::string name() const override { return "json"; }
+
+  std::string Encode(const std::vector<::opencensus::trace::exporter::SpanData>
+                         &spans) override;
+};
+
+// Pretty print JSON encoding
+class PrettyJsonCodec : public MessageCodec {
+ public:
+  const std::string name() const override { return "pretty_json"; }
+
+  std::string Encode(const std::vector<::opencensus::trace::exporter::SpanData>
+                         &spans) override;
+};
+
+class ExportClient {
+ public:
+  virtual void SendMessage(const std::string &msg, size_t size) = 0;
+};
+
+struct ZipkinExporterOptions {
```

Comment:
A lot of this is duplicated as HttpClientOptions in the *.cc file.

Reply:
Removed HttpClient/HttpClientOptions and merged it with ZipkinExporter.  Also moved mostly everything to .cc file.